### ISSUE TITLE
[Bug] BF16 FMHA returns zero for finite extreme negative logits

### ISSUE THREAD

**Issue by akshatvishu** [ROCm/aiter#5086]
## Summary

Several BF16 FMHA paths return zero when every QK score is finite but below an internal softmax maximum seed.

On MI300X, native D64 returns zero for a score of `-1.2676506002282294e30`. The expected output is exactly one because every value element is one. FMHA v3 D128 and D192/V128 also fail for extreme negative scores.

## Environment

* GPU: AMD Instinct MI300X, gfx942
* ROCm: 7.2.3
* Python: 3.12.13
* PyTorch: `2.11.0+gitd0c8b1f`
* AITER: `main` at `878d60d779fa774496fe075175d0e99f1d49eea9`

## Results

| Path | Finite QK score | Expected | Observed |
| --- | ---: | ---: | ---: |
| Native D64 | `-1.2676506002282294e30` | `1` | `0` |
| FMHA v3 D128 | `-1.914088313930279e38` | `1` | `0` |
| FMHA v3 D192/V128 | `-9.50737950171172e29` | `1` | `0` |
| FMHA v3 D192/V128 | `-3.802951800684688e30` | `1` | `0` |

## Cause

The source kernels use `-1e30f` as a softmax maximum seed:

* Native D64: `csrc/kernels/mha_native/fused/pipeline.hpp`
* OPUS D128: `csrc/include/fmha_fwd_hd128_bf16_opus_kernel.hpp`
* OPUS D192/V128: `csrc/include/fmha_fwd_hd192_v128_bf16_opus_kernel.hpp`

A finite score below that seed never becomes the row maximum. Its exponential then underflows to zero. [FlashInfer PR 4401](https://github.com/flashinfer-ai/flashinfer/pull/4401) fixed the same class of error.

Using the lowest finite FP32 value fixes the source kernels without introducing `(-inf) - (-inf)` for fully masked tiles.

FMHA v3 needs a separate fix in its precompiled code objects. One v3 failure occurs above `-1e30`, so source review does not establish the same cause.

## Validation

A focused regression gives this red and green result on gfx942:

```text
main: 4096 / 4096 output elements are 0 instead of 1
fixed: 1 passed
```

The OPUS source cases need the same regression on gfx950. D128 OPUS is opt in through `AITER_ENABLE_FMHA_OPUS=1`. D192/V128 OPUS is enabled by default.
